### ISSUE TITLE
Get Weather location from IP address

### ISSUE THREAD

**Issue by mattroseman** [addy-dclxvi/almighty-dotfiles#3]
Hey, I love the tint2 themes, they look great.

I was playing around with your weather script, and changed it so it would look up my location based on my IP, and then get the weather from that.

```
IP=$(curl -s http://whatismyip.akamai.com/)

# Get longitude and latitude based on ip
GEO_LOCATION=$(jq -r '.loc' <<< $(curl -s "ipinfo.io/${IP}"))
IFS=',' read -ra LONGLAT <<< "$GEO_LOCATION"
LAT="${LONGLAT[0]}"
LON="${LONGLAT[1]}"

WEATHER_URL="http://api.openweathermap.org/data/2.5/weather?lat=${LAT}&lon=${LON}&appid=${API_KEY}&units=imperial"
```

you need **jq** to parse the JSON in the ipinfo.io request.

Thought you might be interested.

**Comment by addy-dclxvi** [comment]
Thank You Mr.Roseman 😊

I'm really interested. Gonna check it after back to keyboard. Still away now.

Cheers!

**Comment by mattroseman** [comment]
I've continued to play with this script, and changed it so that it'll check for internet at a higher frequency, then check for weather once connected.

```
while :
do
    IP=$(curl -s http://whatismyip.akamai.com/)
    if [ -z "$IP" ]; then
        echo ""
    else
        get_weather $1
        sleep 30m
    fi
    sleep 1
done
```

where get_weather is a function I wrapped all the other code in.

I also set tint2 to continuous output with no interval.

This way if there is no internet, itll display nothing, but check every second, and as soon as there is internet it'll show the weather in tint2.
